src/pip/_internal/utils/misc.py

```diff
@@ -907,6 +911,18 @@ def remove_auth_from_url(url):
     return surl
 
 
+def redact_password_from_url(url):
```

Comment:
I would move this function to be after `remove_auth_from_url()` (since it is a "fancier" version).